src/app/text-marking/text-marking.ts

```diff
@@ -101,6 +114,138 @@ export function markText(
+ * @param status - The status indicating whether to 'add' or 'remove' the animation class.
+ */
+function updateHighlightingMarkings(
+    newNode: HTMLElement,
+    status: string
+): void {
+    if (status === 'add') {
+        newNode.classList.add(MARKING_TEXT);
+    } else {
+        newNode.classList.remove(MARKING_TEXT);
+    }
+}
+
+/**
+ * Updates the array of highlighted marking words based on changes in the text content.
+ * This function is called when a node has been added, removed, or modified.
+ *
+ * @param newNode - The HTML element representing the updated node.
+ * @param newTextContent - The new text content of the updated node.
+ */
+function updatedMarkings(newNode: HTMLElement, newTextContent: string): void {
+    // Handle scenarios for additions, removals, and modifications
+    const deletedNodeNum: number =
+        lastHighlightedMarkingWords.length - maxMarkings; // gets the difference between the old state's length and the new one.
+    const addedNodes: number = maxMarkings - lastHighlightedMarkingWords.length; // gets the number of new nodes being added
+
+    if (addedNodes > 0) {
+        // If a new first node was added, insert it and apply animation
+        addItemAtIndex(markingInex, newTextContent);
+        updateHighlightingMarkings(newNode, 'add');
+    } else if (
+        lastHighlightedMarkingWords[markingInex + deletedNodeNum] ===
+        newTextContent
+    ) {
+        // Remove the item if it was changed back to the previous content
+        updateHighlightingMarkings(newNode, 'remove');
+        removeItemAtIndex(markingInex);
+    }
+}
+
+/**
+ * Updates the state of the lastHighlightedMarkingWords array. This function is designed to be
+ * called when a specific condition is met, ensuring that the lastHighlightedMarkingWords array
+ * reflects the current state of the highlightedMarkingWords array.
+ *
+ * This action is typically triggered after the completion of a specific operation or task, ensuring
+ * that the arrays are synchronized for future comparisons.
+ */
+function pushArrayItems(): void {
+    lastHighlightedMarkingWords.length = 0;
+    highlightedMarkingWords.forEach((item) => {
+        lastHighlightedMarkingWords.push(item);
+    });
+    highlightedMarkingWords.length = 0;
+}
+
+/**
+ * Removes an item from the lastHighlightedMarkingWords array at the specified index.
+ * @param {number} index - The index of the item to be removed from the array.
+ */
+function removeItemAtIndex(index: number): void {
+    lastHighlightedMarkingWords.splice(index, 1);
+}
+
+/**
+ * Adds a new item to the lastHighlightedMarkingWords array at the specified index.
+ * @param {number} index - The index at which to insert the new item.
+ * @param {string} newItem - The new item to be added to the array.
+ */
+function addItemAtIndex(index: number, newItem: string): void {
+    lastHighlightedMarkingWords.splice(index, 0, newItem);
```

Comment:
I kind of feel like both of these methods can't really justify their existence, what will happen if we will want to use splice with another argument, should we create another method for that? Personally I would just remove them and inline these lines of code where their respective methods are used.